lib/extension-es6.js

```diff
@@ -7,7 +7,8 @@ function es6(loader) {
 
   var transpiler, transpilerModule, transpilerRuntimeModule, transpilerRuntimeGlobal;
 
-  var isBrowser = typeof window != 'undefined';
+  var isBrowser = typeof window !== 'undefined' && !!window.Window && window instanceof window.Window;
+  var isWorker = typeof self !== 'undefined' && !!self.WorkerGlobalScope && self instanceof WorkerGlobalScope;
```

Comment:
Perhaps lets stick to one condition and call it `autoLoadTranspiler` or something like that?

I'm not sure we need the extra window checks though? Also for the worker check, typeof WorkerGlobalScope != 'undefined' should be enough?